test/python/circuit/test_singleton_gate.py

```diff
@@ -251,3 +253,26 @@ def test_positional_label(self):
         label_gate = SXGate("I am a little label")
         self.assertIsNot(gate, label_gate)
         self.assertEqual(label_gate.label, "I am a little label")
+
+    def test_immutable_pickle(self):
+        gate = SXGate()
+        self.assertFalse(gate.mutable)
+        with io.BytesIO() as fd:
+            pickle.dump(gate, fd)
+            fd.seek(0)
+            copied = pickle.load(fd)
```

Comment:
Totally unimportant, but pickle allows serialisation to `bytes` directly, so you can do this as a one-liner if you prefer: `copied = pickle.loads(pickle.dumps(gate))`.

Reply:
I just don't like using `pickle.dumps` because `s` implies string to me and it's not a string :D

Reply:
maybe it's an initial from a language that reads right-to-left: "DUMP byteS" instead of "DUMP String" lol